### PR TITLE
Submit 2 plugin:hash and kv

### PR DESCRIPTION
**Pull request by glight2000**
hash plugin is used to create md5/sha1... code for the given string.
kv plugin is a key-value store based on sqlite3. It can set/get/del key-value pairs. It will also create db file in the parent directory of this script file.

### CODE REVIEW

File: org.albert.extension.external.hash.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+if albert_op == "METADATA":
+    metadata = """{
+      "iid":"org.albert.extension.external/v2.0",
+      "name":"hash",
+      "version":"1.0",
+      "author":"glight2000",
+      "dependencies":[],
+      "trigger":"hash"
+    }"""
+    print(metadata)
```
Comment (idkCpp) [inline org.albert.extension.external.hash.py]: Please make `metadata` an object and print it with `json.dumps`

File: org.albert.extension.external.kv.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+if albert_op == "METADATA":
+    metadata = """{
+      "iid":"org.albert.extension.external/v2.0",
+      "name":"kv",
+      "version":"1.0",
+      "author":"glight2000",
+      "dependencies":[],
+      "trigger":"kv"
+    }"""
+    print(metadata)
```
Comment (idkCpp) [inline org.albert.extension.external.kv.py]: Here as well...

**Comment by glight2000** [comment]
Replace done. Please review.
Thanks.
2 advice: centralized db like store and a easy debug way~.

File: org.albert.extension.external.kv.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+def filelog(str):
+    # f=open("{0}/../testplugin.log".format(os.path.split(os.path.realpath(__file__))[0]),"a")
+    # f.writelines(str)
+    # f.writelines("\n")
+    # f.close()
+    return
+
+
+albert_op = os.environ.get("ALBERT_OP")
+mode = os.environ.get("MODE")
```
Comment (idkCpp) [inline org.albert.extension.external.kv.py]: What does this?

File: org.albert.extension.external.kv.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+def action(name, command, arguments):
+    action = Object()
+    action.name = name
+    action.command = command
+    action.arguments = arguments
+    return action
+
+
+def createTalbe():
```
Comment (ManuelSchneid3r) [inline org.albert.extension.external.kv.py]: typo

File: org.albert.extension.external.kv.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+def delete(key):
+    cursor = conn.execute("DELETE FROM kv WHERE `key` = '{0}'".format(key))
+    conn.commit()
+    cursor.close()
+    return
+
+
+# debug
+def filelog(str):
```
Comment (ManuelSchneid3r) [inline org.albert.extension.external.kv.py]: Remove debug code in the final version. Further take a look at [Python Logging](https://docs.python.org/2/library/logging.html).

**Comment by ManuelSchneid3r** [comment]
@glight2000 do you plan to work on the requests? I think a general kv extension is a great idea.

File: org.albert.extension.external.hash.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+elif albert_op == "NAME":
+    sys.exit(0)
+
+elif albert_op == "INITIALIZE":
+    sys.exit(0)
+
+elif albert_op == "FINALIZE":
+    sys.exit(0)
+
+elif albert_op == "SETUPSESSION":
```
Comment (ManuelSchneid3r) [inline org.albert.extension.external.hash.py]: SETUPSESSION+TEARDOWNSESSION will be dropped . You can remove them.

File: org.albert.extension.external.kv.py [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+elif albert_op == "INITIALIZE":
+    conn = sqlite3.connect(dbFile)
+    createTalbe()
+    conn.close()
+    sys.exit(0)
+
+elif albert_op == "FINALIZE":
+    sys.exit(0)
+
+elif albert_op == "SETUPSESSION":
```
Comment (ManuelSchneid3r) [inline org.albert.extension.external.kv.py]: Same as above.

**Comment by ManuelSchneid3r** [comment]
Will come as native plugins, thanks for the ideas.

**Comment by glight2000** [comment]
@ManuelSchneid3r @idkCpp  Sorry for haven't been here for days and didn't see the reply. And thanks for reviewing my code and the tips. Native plugins are nice.  Thanks Albert again, it's wonderful.

**Pull request closed without merging**